server/index.js

```diff
@@ -54,6 +54,117 @@ app.get('/api/products/:productId', (req, res, next) => {
 });
 
+app.get('/api/cart', (req, res, next) => {
+  const cartId = req.session.cartId;
+  if (!req.session.cartId) {
+    return [];
+  } else {
+    const sql = `select "c"."cartItemId",
+                "c"."price",
+                "p"."productId",
+                "p"."image",
+                "p"."name",
+                "p"."shortDescription"
+            from "cartItems" as "c"
+            join "products" as "p" using ("productId")
+          where "c"."cartId" = $1
+          `;
+
+    const values = [cartId];
+
+    db.query(sql, values)
+      .then(result => res.status(200).json(result.rows))
+      .catch(err => next(err));
+  }
+});
+
+app.post('/api/cart', (req, res, next) => {
+  const productId = req.body.productId;
+
+  if (!parseInt(productId, 10)) {
+    return res.status(400).json({ error: 'invalid productId' });
+  }
+
+  const sql = `
+    select "price"
+      from "products"
+    where "productId" = $1
+    `;
+
+  const values = [productId];
+
+  db.query(sql, values)
+    .then(result => {
+      const price = result.rows[0].price;
+      const sql = `
+        insert into "carts" ("cartId", "createdAt")
+        values (default, default)
+        returning "cartId"
+      `;
+      if (!result.rows.length === 0) {
+        return (new ClientError(`cannot ${req.method} ${req.originalUrl}`, 404));
+      }
+
+      if (req.session.cartId) {
+        const cartId = req.session.cartId;
+        const queriedCartId = [{ cartId: cartId }];
+        return { queriedCartId, price };
+      } else {
+        return (
+          db.query(sql)
+            .then(result => {
+              const queriedCartId = result.rows;
+              return { queriedCartId, price };
+            })
+        );
+      }
+
+    })
+    .then(result => {
+      const productId = req.body.productId;
+      const price = result.price;
+      const cartIdResult = result.queriedCartId[0].cartId;
+      req.session.cartId = cartIdResult;
+
+      const sql = `
+        insert into "cartItems" ("cartId", "productId", "price")
+        values ($1, $2, $3)
+        returning "cartItemId"
+      `;
+
+      const values = [cartIdResult, productId, price];
+
+      return (
+        db.query(sql, values)
+          .then(result => {
+            const cartItemId = result.rows[0];
+            return cartItemId;
+          })
+      );
+    })
+    .then(result => {
+      const cartItemId = result.cartItemId;
+      const sql = `
+        select "c"."cartItemId",
+              "c"."price",
+              "p"."productId",
+              "p"."image",
+              "p"."name",
+              "p"."shortDescription"
+          from "cartItems" as "c"
+          join "products" as "p" using ("productId")
+        where "c"."cartItemId" = $1
+      `;
+      const values = [cartItemId];
+
+      return (
+        db.query(sql, values)
+          .then(result => res.status(201).json(result.rows[0]));
+      })
+      .catch(err => next(err));
+      );
```

Comment:
Can you format this a little? I can't really tell if it's valid based on the indentation. The overall structure should be 
```js
db.query(firstQuery)
  .then(() => {
    // code code code
  })
  .then(() => {
    // code code code
  })
  .then(() => {
    // code code code
  })
  .catch(err => {
    // handle error
  });
```